dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java

```diff
@@ -231,6 +244,132 @@ public void testconvertRelationshipToRelationshipField_SelfRelated_Success() thr
         }
     }
 
+    /**
+     * <b>Method to test:</b> {@link RelationshipAPI#dbRelatedContent(Relationship, Contentlet)} <p>
+     * <b>Given Scenario:</b> A content is related to another (different content types) <p>
+     * <b>ExpectedResult:</b> {@link RelationshipAPI#dbRelatedContent(Relationship, Contentlet)} should always return related content
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void testDbRelatedContentWhenNonSelfJoinRelationship()
+            throws DotDataException, DotSecurityException {
+        final ContentType blogContentType = TestDataUtils.getBlogLikeContentType();
+        final ContentType commentContentType = TestDataUtils.getCommentsLikeContentType();
+
+        final Field field = FieldBuilder.builder(RelationshipField.class)
+                .name(commentContentType.variable())
+                .contentTypeId(blogContentType.id())
+                .values(String.valueOf(RELATIONSHIP_CARDINALITY.MANY_TO_MANY.ordinal()))
+                .relationType(commentContentType.variable()).required(false).build();
+
+        contentTypeFieldAPI.save(field, user);
+
+        //Creates parent content
+        Contentlet blogContentParent = new ContentletDataGen(blogContentType.id())
+                .languageId(APILocator.getLanguageAPI().getDefaultLanguage().getId())
+                .setProperty("title", "blogContentParent")
+                .setProperty("urlTitle", "blogContentParent")
+                .setProperty("author", "systemUser")
```

Comment:
Codacy found an issue: [The String literal "author" appears 4 times in this file; the first occurrence is on line 273](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)